searx/botdetection/install.py

```diff
@@ -6,33 +6,38 @@
 import sys
 import flask
 
-from searx import redisdb
-from searx.plugins import logger
+from searx import redisdb, logger
 from searx.botdetection import limiter
 
-name = "Request limiter"
-description = "Limit the number of request"
-default_on = False
-preference_section = 'service'
-
+# the configuration are limiter.toml and "limiter" in settings.yml
+# so, for coherency, the logger is "limiter" even if the module name "searx.botdetection"
 logger = logger.getChild('limiter')
 
 
+_INSTALLED = False
+
+
 def pre_request():
     """See :ref:`flask.Flask.before_request`"""
     return limiter.filter_request(flask.request)
 
 
-def init(app: flask.Flask, settings) -> bool:
+def is_installed():
+    return _INSTALLED
+
+
+def initialize(app: flask.Flask, settings):
+    """Instal the botlimiter aka limiter"""
+    global _INSTALLED  # pylint: disable=global-statement
     if not settings['server']['limiter'] and not settings['server']['public_instance']:
-        return False
+        return
```

Comment:
This and what the undocumented `is_installed()` returns is unclear to me.

Does it mean `is_installed()` returns true only when redis is available and `settings['server']['public_instance']` is set to `true`?

Sorry, but this option `public_instance` only causes confusion, how is a developer supposed to understand this code without getting all the undocumented background to it?